### PR TITLE
Feature/145 filter by node

### PR DESCRIPTION
**Pull request by vugaragha**

### CODE REVIEW

File: api/src/main/java/de/valtech/aecu/api/groovy/console/bindings/filters/FilterByNode.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2018 - 2019 Valtech GmbH
```
Comment (gruberrolandvaltech) [inline api/src/main/java/de/valtech/aecu/api/groovy/console/bindings/filters/FilterByNode.java:2]: 2021

File: api/src/test/java/de/valtech/aecu/core/groovy/console/bindings/filters/FilterByNodeNotExistTest.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2018 Valtech GmbH
```
Comment (gruberrolandvaltech) [inline api/src/test/java/de/valtech/aecu/core/groovy/console/bindings/filters/FilterByNodeNotExistTest.java:2]: 2021

File: api/src/main/java/de/valtech/aecu/api/groovy/console/bindings/filters/FilterByNode.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+     * @param path node path
+     */
+    public FilterByNode(@Nonnull String path) {
+        this.path = path;
+    }
+
+    @Override
+    public boolean filter(@Nonnull Resource resource, StringBuilder output) {
+        if(StringUtils.isBlank(path))
+            return true;
```
Comment (gruberrolandvaltech) [inline api/src/main/java/de/valtech/aecu/api/groovy/console/bindings/filters/FilterByNode.java:49]: use brackets for all if-blocks
recheck if the formatter rules were correctly imported. There should be a space after the "if"

File: api/src/main/java/de/valtech/aecu/api/groovy/console/bindings/filters/FilterByNodeNotExist.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+import javax.annotation.Nonnull;
+
+
+/**
+ * Filters resources by node. Only resources that do not exist in the repository are accepted.
+ *
+ * @author Vugar Aghayev
+ */
+public class FilterByNodeNotExist implements FilterBy {
```
Comment (gruberrolandvaltech) [inline api/src/main/java/de/valtech/aecu/api/groovy/console/bindings/filters/FilterByNodeNotExist.java:33]: The negative filter case can be handled by same FilterByNode class using a parameter.
No need to duplicate the code.

File: api/src/test/java/de/valtech/aecu/core/groovy/console/bindings/filters/FilterByNodeTest.java [hunk 5]
```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2018 Valtech GmbH
```
Comment (gruberrolandvaltech) [inline api/src/test/java/de/valtech/aecu/core/groovy/console/bindings/filters/FilterByNodeTest.java:2]: 2021

**Review by gruberrolandvaltech — CHANGES_REQUESTED** [review]

File: api/src/main/java/de/valtech/aecu/api/groovy/console/bindings/filters/FilterByNode.java (hunk 1 above)
Comment (vugaragha) [reply api/src/main/java/de/valtech/aecu/api/groovy/console/bindings/filters/FilterByNode.java]: Done

File: api/src/test/java/de/valtech/aecu/core/groovy/console/bindings/filters/FilterByNodeNotExistTest.java (hunk 2 above)
Comment (vugaragha) [reply api/src/test/java/de/valtech/aecu/core/groovy/console/bindings/filters/FilterByNodeNotExistTest.java]: Done

File: api/src/main/java/de/valtech/aecu/api/groovy/console/bindings/filters/FilterByNode.java (hunk 3 above)
Comment (vugaragha) [reply api/src/main/java/de/valtech/aecu/api/groovy/console/bindings/filters/FilterByNode.java:52]: imported the Intellij formatter, now the code is formatted as expected.

File: api/src/test/java/de/valtech/aecu/core/groovy/console/bindings/filters/FilterByNodeTest.java (hunk 5 above)
Comment (vugaragha) [reply api/src/test/java/de/valtech/aecu/core/groovy/console/bindings/filters/FilterByNodeTest.java]: Done

File: api/src/main/java/de/valtech/aecu/api/groovy/console/bindings/filters/FilterByNodeNotExist.java (hunk 4 above)
Comment (vugaragha) [reply api/src/main/java/de/valtech/aecu/api/groovy/console/bindings/filters/FilterByNodeNotExist.java]: parameterized FilterByNode, the duplicated code has been removed.

**Pull request merged**